Jenkinsfile

```diff
@@ -78,130 +77,123 @@ pipeline {
                     }
                 }
-            }
-        }
 
-        stage('Build') {
-            agent { label 'builder' }
-            environment {
-                THIRD_PARTY_CACHE_PATH = "C:/BUILD_DEPENDENCIES/third_party_cache/"
-                DRIVER_SRC_PATH = "github.com/Juniper/contrail-windows-docker-driver"
-                BUILD_IN_RELEASE_MODE = "false"
-                SIGNTOOL_PATH = "C:/Program Files (x86)/Windows Kits/10/bin/x64/signtool.exe"
-                CERT_PATH = "C:/BUILD_DEPENDENCIES/third_party_cache/common/certs/codilime.com-selfsigned-cert.pfx"
-                CERT_PASSWORD_FILE_PATH = "C:/BUILD_DEPENDENCIES/third_party_cache/common/certs/certp.txt"
-                COMPONENTS_TO_BUILD = "DockerDriver,Extension,Agent"
-
-                MSBUILD = "C:/Program Files (x86)/MSBuild/14.0/Bin/MSBuild.exe"
-                WINCIDEV = credentials('winci-drive')
-            }
-            steps {
-                deleteDir()
+                stage('Build') {
+                    agent { label 'builder' }
+                    environment {
+                        THIRD_PARTY_CACHE_PATH = "C:/BUILD_DEPENDENCIES/third_party_cache/"
+                        DRIVER_SRC_PATH = "github.com/Juniper/contrail-windows-docker-driver"
+                        BUILD_IN_RELEASE_MODE = "false"
+                        SIGNTOOL_PATH = "C:/Program Files (x86)/Windows Kits/10/bin/x64/signtool.exe"
+                        CERT_PATH = "C:/BUILD_DEPENDENCIES/third_party_cache/common/certs/codilime.com-selfsigned-cert.pfx"
+                        CERT_PASSWORD_FILE_PATH = "C:/BUILD_DEPENDENCIES/third_party_cache/common/certs/certp.txt"
+                        COMPONENTS_TO_BUILD = "DockerDriver,Extension,Agent"
+
+                        MSBUILD = "C:/Program Files (x86)/MSBuild/14.0/Bin/MSBuild.exe"
+                        WINCIDEV = credentials('winci-drive')
+                    }
+                    steps {
+                        deleteDir()
 
-                unstash "CIScripts"
-                unstash "SourceCode"
+                        unstash "CIScripts"
+                        unstash "SourceCode"
 
-                powershell script: './CIScripts/BuildStage.ps1'
+                        powershell script: './CIScripts/BuildStage.ps1'
 
-                stash name: "Artifacts", includes: "output/**/*"
-            }
-            post {
-                always {
-                    deleteDir()
+                        stash name: "Artifacts", includes: "output/**/*"
+                    }
+                    post {
+                        always {
+                            deleteDir()
+                        }
+                    }
                 }
-            }
-        }
 
-        stage('Cleanup-Provision-Deploy-Test') {
-            agent none
-            when { environment name: "DONT_CREATE_TESTBEDS", value: null }
+                stage('Cleanup-Provision') {
```

Comment:
I think better name would be just "Provision". It is going to look better in Jenkins. I will a comment that cleanup is done before provisioning.